test/integration/camera_sensor.cc

```diff
@@ -2040,4 +2040,230 @@ TEST_F(CameraSensor, SetCompositorNames)
+      setPose.Rot().Euler(), width, height, updateRate,
+      "", 0, 0, true, -0.5, 0, 0, 0, 0, 0.5, 0.5,
+      false, horizontalFov);
+
+  sensors::SensorPtr sensorPincushionLegacy =
+    sensors::get_sensor(cameraNamePincushionLegacy);
+  sensors::CameraSensorPtr camSensorPincushionLegacy =
+    std::dynamic_pointer_cast<sensors::CameraSensor>(sensorPincushionLegacy);
+  sensors::SensorPtr sensorPincushionNew =
+    sensors::get_sensor(cameraNamePincushionNew);
+  sensors::CameraSensorPtr camSensorPincushionNew =
+    std::dynamic_pointer_cast<sensors::CameraSensor>(sensorPincushionNew);
+
+  sensors::SensorPtr sensorBarrelLegacy =
+    sensors::get_sensor(cameraNameBarrelLegacy);
+  sensors::CameraSensorPtr camSensorBarrelLegacy =
+    std::dynamic_pointer_cast<sensors::CameraSensor>(sensorBarrelLegacy);
+  sensors::SensorPtr sensorBarrelNew =
+    sensors::get_sensor(cameraNameBarrelNew);
+  sensors::CameraSensorPtr camSensorBarrelNew =
+    std::dynamic_pointer_cast<sensors::CameraSensor>(sensorBarrelNew);
+
+  imageCount = 0;
+  imageCount2 = 0;
+  imageCount3 = 0;
+  imageCount4 = 0;
+
+  img = new unsigned char[width * height*3];
+  img2 = new unsigned char[width * height*3];
+  img3 = new unsigned char[width * height*3];
+  img4 = new unsigned char[width * height*3];
+
+  event::ConnectionPtr c1 =
+    camSensorPincushionLegacy->Camera()->ConnectNewImageFrame(
+        std::bind(&::OnNewCameraFrame, &imageCount, img,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+  event::ConnectionPtr c2 =
+    camSensorPincushionNew->Camera()->ConnectNewImageFrame(
+        std::bind(&::OnNewCameraFrame, &imageCount2, img2,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+  event::ConnectionPtr c3 =
+    camSensorBarrelLegacy->Camera()->ConnectNewImageFrame(
+        std::bind(&::OnNewCameraFrame, &imageCount3, img3,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+  event::ConnectionPtr c4 =
+    camSensorBarrelNew->Camera()->ConnectNewImageFrame(
+        std::bind(&::OnNewCameraFrame, &imageCount4, img4,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+
+  // Get some images
+  while (
+    imageCount < 10 || imageCount2 < 10 ||
+    imageCount3 < 10 || imageCount4 < 10)
+  {
+    common::Time::MSleep(10);
+  }
+
+  // Compare colors. Barrel distorted image should have more darker pixels than
+  // the original as the ground plane has been warped to occupy more of the
+  // image. The same should be true for pincushion distortion, because the
+  // ground plane is still distorted to be larger - just different parts
+  // of the image are distorted.
```

Comment:
I agree with the content of this comment block, but we aren't actually testing this, since we don't have an undistorted image to compare against